src/test_embedding.py

```diff
@@ -0,0 +1,278 @@
+"""
+Hydra-based Embedding Extractor (Direct JSON)
+=============================================
+This version skips the HDF5/DataModule pipeline and directly
+builds graph features from JSON discussion trees for embedding extraction.
+
+Usage:
+------
+python src/test_embedding.py \
+  --json src/test_data.json \
+  --ckpt src/logs/pretrain/runs/2025-10-23_22-41-28/checkpoints/last.ckpt \
+  --device cuda \
+
+"""
+
+# the batch size command is removed as it actually valids the parameters,
+# need to see why
+
+import os
+import sys
+import json
+import torch
+import argparse
+import numpy as np
+from omegaconf import OmegaConf
+from hydra.utils import instantiate
+from transformers import AutoTokenizer, AutoImageProcessor
+from PIL import Image
+from data.types import GraphFeatures, TextFeatures, ImageFeatures
+from data import collator_utils
+from tasks import dataset_utils as dut
+
+# ---- Pathing ----
+ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.extend([ROOT, os.path.join(ROOT, "src")])
+
+
+#  Convert a tree jsn to feature dicts, the graph
+#  watch: Similar to data.dataset_utils.tree_to_graph_features but simplified
+def tree_to_graph_features(tree, text_tokenizer, image_processor, root_dir="."):
+    """Flatten a discussion tree into tensors for model input."""
+    dut.compute_relative_distance(tree)
+
+    flat = {
```

Comment:
We are missing a few parameters here. Namely, `is_root` and `distances`. These should be already attributes of node (I think). 

https://github.com/liamhebert/MDT-2/blob/5f50b4abbf32d626f9695617751a38b778e3f224/src/tasks/dataset.py#L860-L867